providers/refresh.md

```diff
@@ -0,0 +1,140 @@
+
+Once here the refresh of a target consists of three steps:
+1. Collection of inventory from the native provider: [`#collect_inventory_for_targets`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L84)
+2. Parsing of the inventory into the MIQ schema: [`#parse_targeted_inventory`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L92)
+3. Saving inventory to the database: [`#save_inventory`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L96)
+
+As a provider author you have the ability to override any of these methods if you wish, but by default we will invoke the helper classes generated by the provider generator.  These classes are under the `ManageIQ::Provider::#{vendor}::Inventory` namespace and map to the three components of refresh: collect, parse, save.
+
+###### Inventory::Collector
+
+A collector encapsulates how inventory is retrieved from the management system and is typically the only part of a refresh that issues API calls.  At first this can seem like extra overhead as the parser could just as easily also collect inventory, but this becomes important when also implementing targeted refresh.  By controlling what is returned from the collector you can typically reuse the Parsers for full and targeted refresh.
+
+For example your "full" collector might look like:
+```ruby
+class ManageIQ::Providers::MyProvider::Inventory::Collector::CloudManager
+  def vms
+    connection.all_vms
+  end
+end
+```
+
+And your "targeted" collector:
+```ruby
+class ManageIQ::Providers::MyProvider::Inventory::Collector::TargetCollection
+  def vms
+    targets.collect { |target| connection.get_vm(target.instance_id) }
+  end
+end
+```
+
+This way your parser can call `collector.vms` without having to care if it is performing a full or a targeted refresh.
+
+###### Inventory::Persister
+
+It might seem a little out of order to talk about the persister before the parser, but a lot of concepts here are required to understand how the parser works.
+
+The persister class defines the set of inventory collections included in the refresh.  An inventory collection defines how records are saved to the database.
+
+Typically inventory collections map to models/database tables for example the `vms` collection saves records to the `"vms"` table.  The inventory collections will contain an array of `InventoryObject` which will become the records which are created in that table.
+
+The persister also defines what will be deleted as part of a refresh.  The saver will delete the compliment of what is in the inventory collection based on a certain scope.  During a full refresh this scope is the entire association (e.g. the equivalent of `ems.vms.all`) and during a targeted refresh the scope is basically the set of targets.
```

Comment:
Assuming you meant "inverse", then this is a typo.

```suggestion
The persister also defines what will be deleted as part of a refresh.  The saver will delete the complement of what is in the inventory collection based on a certain scope.  During a full refresh this scope is the entire association (e.g. the equivalent of `ems.vms.all`) and during a targeted refresh the scope is basically the set of targets.
```

Reply:
It compliments it first, then deletes it :laughing: